fog-updater/src/fog_update.py

```diff
@@ -0,0 +1,252 @@
+                state = None
+
+        print(line, file=output)
+
+    return output.getvalue()
+
+
+def get_latest_metrics_index():
+    r = requests.get(INDEX_URL)
+    r.raise_for_status()
+    return r.text
+
+
+def _rewrite_repositories_yaml(repo, branch, data):
+    contents = repo.get_contents("repositories.yaml", ref=branch)
+    content = contents.decoded_content.decode("utf-8")
+
+    new_content = content
+    for item in data:
+        name, metrics_or_pings, library, files = item
+        new_content = swap_file_list(
+            new_content, name, files, metrics_or_pings, library
+        )
+
+    if content == new_content:
+        raise Exception(
+            "Update to repositories.yaml resulted in no changes: maybe the file was already up to date?"  # noqa
+        )
+
+    return new_content
+
+
+def _commit_repositories_yaml(repo, branch, author, new_content):
+    contents = repo.get_contents("repositories.yaml", ref=branch)
+
+    repo.update_file(
+        contents.path,
+        "Update repositories.yaml with new FOG metrics_yamls list",
+        new_content,
+        contents.sha,
+        branch=branch,
+        author=author,
+    )
+
+    return True
+
+
+def main(argv, repo, author, debug=False, dry_run=False):
+    if len(argv) < 1:
+        print(USAGE)
+        sys.exit(1)
+
+    release_branch_name = "main"
+    short_version = "main"
+
+    metrics_index = get_latest_metrics_index()
+    data = eval_extract(metrics_index)
+    gecko_metrics = sorted(data["gecko_metrics"])
+    gecko_pings = sorted(data["gecko_pings"])
+    firefox_desktop_metrics = sorted(data["firefox_desktop_metrics"])
+    firefox_desktop_pings = sorted(data["firefox_desktop_pings"])
+    background_update_metrics = sorted(data["background_update_metrics"])
+    background_update_pings = sorted(data["background_update_pings"])
+
+    data = [
+        # Name, metrics/pings, library?, files
+        ["gecko", "metrics", True, gecko_metrics],
+        ["gecko", "pings", True, gecko_pings],
+        ["firefox_desktop", "metrics", False, firefox_desktop_metrics],
+        ["firefox_desktop", "pings", False, firefox_desktop_pings],
+        ["firefox_desktop_background_update", "metrics", False, background_update_metrics],
+        ["firefox_desktop_background_update", "pings", False, background_update_pings],
+    ]
+
+    print(f"{ts()} Updating repositories.yaml")
+    try:
+        new_content = _rewrite_repositories_yaml(repo, release_branch_name, data)
+    except Exception as e:
+        print(f"{ts()} {e}")
+        return
+
+    if dry_run:
+        print(f"{ts()} Dry-run so not continuing.")
+        return
+
+    # Create a non unique PR branch name for work on this ac release branch.
+    pr_branch_name = f"fog-update/update-metrics-index-{short_version}"
+
+    try:
+        pr_branch = repo.get_branch(pr_branch_name)
+        if pr_branch:
+            print(f"{ts()} The PR branch {pr_branch_name} already exists. Exiting.")
```

Comment:
So this means if we aren't quick enough to review these each day, the work will pile up?

Reply:
... and we wouldn't even hear about it until we merge & delete and the next run comes around!

I'm ok with that for now.
We could switch to adding additional commits to the branch if it exists or always creating a fresh branch or something.